src/libs/DebtQueue.sol

```diff
@@ -0,0 +1,114 @@
+// SPDX-License-Identifier: MIT
+// OpenZeppelin Contracts (last updated v4.6.0) (utils/structs/DoubleEndedQueue.sol)
+pragma solidity ^0.8.4;
+
+import 'openzeppelin-contracts/contracts/utils/math/SafeCast.sol';
+
+/**
+ * @notice Modified version of Open Zeppelin's DoubleEndedQueue, but using `DebtEntry` instead of `bytes32` for the values in the queue
+ * DoubleEndedQueue: https://github.com/OpenZeppelin/openzeppelin-contracts/blob/v4.7.3/contracts/utils/structs/DoubleEndedQueue.sol
+ * The purpose of the queue is to keep track of accounts holding debt and their order
+ * @dev The queue is intended to be used only as FIFO, so the functions `popBack`, `pushFront`, `back`, `clear` were removed
+ */
+library DebtQueue {
+    /**
+     * @dev An operation (e.g. {front}) couldn't be completed due to the queue being empty.
+     */
+    error Empty();
+
+    /**
+     * @dev An operation (e.g. {at}) couldn't be completed due to an index being out of bounds.
+     */
+    error OutOfBounds();
+
+    struct DebtEntry {
+        address account;
+        uint256 amount;
```

Comment:
Considering all practical purposes, we should be able to drop `amount` to `uint96` and save ourselves 20K gas when registering debt.

Reply:
Good idea.
Even with an 18 decimal token it gives us a max value of 2^96/10^18 = ~80e9